lib/grandpa/message.go

```diff
@@ -77,21 +77,55 @@ func (v *VoteMessage) ToConsensusMessage() (*ConsensusMessage, error) {
 	}, nil
 }
 
-// NeighbourMessage represents a network-level neighbour message
-type NeighbourMessage struct {
-	Version byte
-	Round   uint64
-	SetID   uint64
-	Number  uint32
+// VersionedNeighbourMessage represents the enum of nighbor messages
+type VersionedNeighbourMessage scale.VaryingDataType
+
+// Index Returns VDT index
+func (VersionedNeighbourMessage) Index() uint { return 2 }
+
+func newVersionedNighborMessage() VersionedNeighbourMessage {
+	vdt := scale.MustNewVaryingDataType(V1NeighbourMessage{})
+
+	return VersionedNeighbourMessage(vdt)
+}
+
+// Set updates the current VDT value to be `val`
+func (vnm *VersionedNeighbourMessage) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*vnm)
+	err = vdt.Set(val)
+	if err != nil {
+		return
+	}
+	*vnm = VersionedNeighbourMessage(vdt)
+	return
+}
+
+// Value returns the current VDT value
+func (vnm *VersionedNeighbourMessage) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*vnm)
+	return vdt.Value()
+}
+
+// V1NeighbourMessage represents a network-level neighbour message
+type V1NeighbourMessage struct {
```

Comment:
```suggestion
type V1NeighbourPacket struct {
```
Or even just `NeighbourPacket`

Reply:
supernit maybe `NeighbourPacketV1` would be nicer than `V1NeighbourPacket` for autocompletion purposes